a demo library for calling a http login api

xavierchow opened the pull request:
@makara 
A simple demo as the PR. The code is a little, but what i read/write/think in the journey of learning new language and problem solving is more valuable. I recorded it for sharing.
(https://github.com/xavierchow/go-test/wiki/Memo-and-newbie-guide-for-Go-lang)

<!---
@huboard:{"order":1.0,"milestone_order":1,"custom_state":""}
-->

xavierchow committed:
initial

xavierchow committed:
comment proxy

xavierchow committed:
Merge remote-tracking branch 'origin/master'

xavierchow committed:
fix a undefined bug

xavierchow committed:
Update README.md

makara commented:
Thanks @xavierchow . We'll get back to you soon.

Meanwhile, I got a question for you:
- The thing you pushed here is usually used in an environment without cookie support right? How do you keep the user logged in for the sub-sequential API requests?

xavierchow commented:
@makara 

> The thing you pushed here is usually used in an environment without cookie support right? How do you keep the user logged in for the sub-sequential API requests?

Yes, that's why I left a TODO comment in my [server code](https://github.com/xavierchow/backend-test/blob/master/index.js#L25), since we have not decide how to access the resource protected by the authentication. There are some approaches commonly used:
- http basic/digest way
- token( cookie or bear token)

An awesome detail explanation is [here](https://stackoverflow.com/questions/319530/restful-authentication/7158864#7158864)
As I implemented the login with session enabled via cookie, I added a test case as an example [here](https://github.com/xavierchow/backend-test/blob/master/test/test-login.js#L45-L57),  Yes the client may not be browser, but we can manipulate the cookie at the same way as browser does.

quentinberder commented:
Done, closing.
